jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -117,23 +117,35 @@ def _param_units(self, param, model_type=None):
     def _update_parameters_from_fit(self):
         """Insert the results of the model fit into the component_models"""
         for m in self.component_models:
+            submodel_index = None
             name = m["id"]
             if hasattr(self._fitted_model, "submodel_names"):
-                if name in self._fitted_model.submodel_names:
-                    m_fit = self._fitted_model[name]
-                else:
+                for i in range(len(self._fitted_model.submodel_names)):
+                    if name == self._fitted_model.submodel_names[i]:
+                        m_fit = self._fitted_model[name]
+                        submodel_index = i
+                if submodel_index is None:
                     continue
             elif self._fitted_model.name == name:
                 m_fit = self._fitted_model
             else:
                 # then the component was not in the fitted model
                 continue
+
             temp_params = []
             for i in range(0, len(m_fit.parameters)):
                 temp_param = [x for x in m["parameters"] if x["name"] ==
                               m_fit.param_names[i]]
                 temp_param[0]["value"] = m_fit.parameters[i]
+                # The submodels don't have uncertainties attached, only the compound model
+                if self._fitted_model.stds is not None:
+                    std_name = temp_param[0]["name"]
+                    if submodel_index is not None:
+                        std_name = f"{std_name}_{submodel_index}"
+                    temp_param[0]["std"] = self._fitted_model.stds[std_name]
```

Comment:
this block confused me... what situation triggers this logic and what does it do?

Reply:
So, this is because in `specutils` the `stds` attribute is only populated on the top-level compound model, not in each of the component models, and they're labeled things like `amplitude_0`, `amplitude_1`. So we have to see which model component is 0, 1, etc, from the order of the `submodel_names` attribute in order to retrieve the correct uncertainties. It's a bit of a pain but fixing it would require some upstream refactoring.